File: 프로그래머스/0/120850. 문자열 정렬하기 （1）/문자열 정렬하기 （1）.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int* compare(const int* a, const int* b){
    return (*a - *b);
}

int* solution(const char* my_string) {
    int* answer = (int*)malloc(strlen(my_string) * sizeof(int));
    int len = 0;
    
    for(int i=0;i<strlen(my_string);i++){
        if(isdigit(my_string[i])){
            answer[len++] = my_string[i] - '0';
        }
    }
    
    qsort(answer, len, sizeof(int), compare);
    
    return answer;
}

File: 프로그래머스/0/120850. 문자열 정렬하기 （1）/문자열 정렬하기 （1）.java
import java.util.*;

class Solution {
    public int[] solution(String my_string) {
        ArrayList<Integer> list = new ArrayList<>();
        
        for(int i=0;i<my_string.length();i++){
            if(Character.isDigit(my_string.charAt(i))){
                list.add(my_string.charAt(i) - '0');
            }
        }
        
        int[] answer = new int[list.size()];
        int len = 0;
        Collections.sort(list); //integer이면 Arrays.sort를 사용하고 ArrayList<Integer> 이면 Collections.sort를 사용한다.
        for(int a:list){
            answer[len++] = a;
        }
        
        return answer;
    }
}